mathesar/models.py

```diff
@@ -309,6 +322,31 @@ def add_constraint(self, constraint_type, columns, name=None):
         return Constraint.objects.create(oid=constraint_oid, table=self)
 
 
+class Column(ReflectingObject):
+    table = models.ForeignKey('Table', on_delete=models.CASCADE, related_name='columns')
+    attnum = models.IntegerField()
+    display_options = JSONField(null=True)
+
+    def __str__(self):
+        return f"{self.__class__.__name__}: {self.table_id}-{self.attnum}"
+
+    def __getattribute__(self, name: str) -> Any:
+        try:
+            return super().__getattribute__(name)
+        except AttributeError:
+            return getattr(self._sa_column, name)
+
+    @cached_property
+    def _sa_column(self):
+        return self.table.sa_columns[self.name]
```

Comment:
Won't this error if the underlying DB table is deleted?

Reply:
@silentninja This question appears unaddressed.

Reply:
I can't seem to get the actual API to error in this case, so I'll mark it resolved.